core/src/main/java/org/elasticsearch/index/get/ShardGetService.java

```diff
@@ -350,6 +355,16 @@ private GetResult innerGetLoadFromStoredFields(String type, String id, String[]
             }
         }
 
+        if (docMapper.parentFieldMapper().active()) {
+            String parentId = ParentFieldSubFetchPhase.getParentId(docMapper.parentFieldMapper(), docIdAndVersion.context.reader(), docIdAndVersion.docId);
+            if (parentId != null) {
```

Comment:
this shouldn't happen. If the parent field is active then a parent id has to be specified, so upon retrieval it should be there.  I'll add an assert here for this.